test/framework/src/main/java/org/opensearch/index/RandomCreateIndexGenerator.java

```diff
@@ -112,8 +114,12 @@ public static void randomMappingFields(XContentBuilder builder, boolean allowObj
         builder.startObject("properties");
 
         int fieldsNo = randomIntBetween(0, 5);
+        Set<String> uniqueFields = new HashSet<>();
         for (int i = 0; i < fieldsNo; i++) {
-            builder.startObject(randomAlphaOfLength(5));
+            uniqueFields.add(randomAlphaOfLength(5));
```

Comment:
same here

Reply:
Simple change to a while loop preserves size.